src/test/java/com/google/sps/PlayerDatabaseTest.java

```diff
@@ -0,0 +1,191 @@
+package com.google.sps;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.sps.data.Player;
+import com.google.sps.data.PlayerDatabase;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests the PlayerDatabase to ensure that it puts, updates, and returns players correctly */
+@RunWith(JUnit4.class)
+public final class PlayerDatabaseTest {
+  // Entity query strings from PlayerDatabase
+  private static final String ENTITY_QUERY_STRING = "player";
+  private static final String DISPLAY_NAME_QUERY_STRING = "displayName";
+  private static final String EMAIL_QUERY_STRING = "email";
+  private static final String ID_QUERY_STRING = "id";
+  private static final String IMAGE_ID_QUERY_STRING = "imageID";
+  private static final String CURRENT_PAGE_ID_QUERY_STRING = "currentPageID";
+  // Mock current user for testing
+  private static final String NAME = "Bob";
+  private static final String NEW_NAME = "Bob2.0";
+  private static final String EMAIL = "Bob@email.com";
+  private static final String AUTH_DOMAIN = "email.com";
+  private static final String CURR_USER_ID = "testid";
+  private static final String IMAGE_ID = "imageId";
+  private static final String NEW_IMAGE_ID = "newImageId";
+  private static final String GAME_STAGE_ID = "gameStageId";
+  private static final String NEW_GAME_STAGE_ID = "newGameStageId";
+  private static Map<String, Object> USER_ID_CONFIG = new HashMap<>();
+
+  static {
+    USER_ID_CONFIG.put("com.google.appengine.api.users.UserService.user_id_key", CURR_USER_ID);
+  }
```

Comment:
Why can't this be put in the setup method?

Reply:
It is used in the constructor of helper, which is declared as a member variable.